doc/source/cuds_definition.rst

```diff
@@ -0,0 +1,89 @@
+CUDS Data Structures
+====================
+
+The Common Unified Data Structures (CUDS) define the expected
+interface for the modelling engine wrappers, the visualisation api the
+top level CUDS containers (i.e. Mesh, Lattice and Particles) and the
+low level CUDS components (e.g Point, Bond, LatticeNode)
+
+.. note::
+
+   The CUDS API is constantly evolving and changes through the processes
+   of the SSB and the designated workgroups.
+
+
+Modelling engine
+----------------
+
+.. figure:: ./images/engine.png
+
+  **Figure. 1:** UML description of a CUDS modelling engine.
+
+The CUDS engine supports operations to add, get and remove CUDS
+containers from the internal memory of the engine wrapper to setup
+the SD state of the simulations.
+
+When a CUDS container is added the CUDS modelling engine will copy
+into internal memory only the ``uid`` mapping and the CUBA information
+that it can support. The user needs to refer to the wrapper package
+to identify what information is retained.
+
+When a CUDS container is returned as a result of a ``get`` operation
+in a modelling engine the returned Container is a proxy to the
+information stored in the internal memory if the wrapper/modelling
+engine.
+
+CUDS containers
+---------------
+
+.. figure:: ./images/containers.png
+
+  **Figure. 2:** UML description of a CUDS Containers
+
+The CUDS containers, with the exception of the Lattice, are multi-type
+container (i.e. contain different types of objects). For each type the
+following operations are currently supported:
+
+* Add an item to the container
+* Get an item from the container based on the ``uid``
+* Remove an item from the container based on the ``uid``
+* Update an item
+* Iterate over all or some of the items given an iterable of ``uids``
+
+.. note::
+
+   For Lattice items do not have a ``uid`` but are accessed based on
+   ``(i,j,k)`` index tuples.
+
+.. rubric:: Snapshot principle
+
+All CUDS containers (native or proxy based) are owners of their data
+and will always return cuds components that container copy (snapshot)
+of the internal data representation. As a result running comparing the
```

Comment:
Suggestion:  The "As a result.." sentence could be changed to  "As a result, the information of an item extracted with a ``get`` operation before and and after a simulation run is not expected to be the same."